### ISSUE TITLE
Can't set dynamic meta tags with useSeoMeta

### ISSUE THREAD

**Issue by carlosvaldesweb** [nuxt/nuxt#26808]
### Environment

- Operating System: Darwin
- Node Version:     v21.7.1
- Nuxt Version:     3.11.1
- CLI Version:      3.11.0
- Nitro Version:    2.9.4
- Package Manager:  npm@10.5.0
- Builder:          -
- User Config:      css, extends, modules, stripe, eslint, image, runtimeConfig, imports, ui, devtools, experimental, colorMode
- Runtime Modules:  @nuxt/ui@2.15.0, @nuxt/image@1.4.0, @nuxtjs/eslint-module@4.1.0, @vueuse/motion/nuxt@2.1.0, nuxt-clarity-analytics@0.0.6, dayjs-nuxt@2.1.9, @unlok-co/nuxt-stripe@2.0.0, @nuxtjs/i18n@8.2.0
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/nuxt-starter-jay43e?file=pages%2Findex.vue

### Describe the bug

I'm trying to use useSeoMeta to take advantage the benefits of typescript instead of use definePageMeta, but i want to have the title as ogTitle and twitterTitle, and description as ogDescription and twitterDescriptio but it doesnt works.

**Comment by danielroe** [comment]
You do it like this:

```ts
useSeoMeta({
  ogTitle: () => route.meta.title,
  ogDescription: () => route.meta.description,
  twitterTitle: () => route.meta.title,
  twitterDescription: () => route.meta.description,
});
```

**Comment by carlosvaldesweb** [comment]
> You do it like this:
> 
> ```ts
> useSeoMeta({
>   ogTitle: () => route.meta.title,
>   ogDescription: () => route.meta.description,
>   twitterTitle: () => route.meta.title,
>   twitterDescription: () => route.meta.description,
> });
> ```

I've updated my stackblitz but it does not works

**Comment by danielroe** [comment]
It's working fine:
![CleanShot 2024-04-25 at 09.20.55@2x.png](https://volta.s3.fr-par.scw.cloud/Clean_Shot_2024_04_25_at_09_20_55_2x_f68a323a58.png)

Please note that `route.meta` is not the same thing as `useSeoMeta`. Route metadata is set using `definePageMeta` and is not related to `useSeoMeta`.

**Comment by carlosvaldesweb** [comment]
@danielroe Yes it's working, but only for title and description. What about of og and twitter tags? I mean that i want to use the title provided in index.vue in the og tags and the twitter tags in app.vue. Where are the twitter tags? That's what I was referring to from the beginning, that I want to use useSeoMeta to take advantage of typescript but assign what comes in the title to the twitter and og tags.

**Comment by danielroe** [comment]
What are you trying?

**Comment by Jesus82** [comment]
@danielroe I have the same user case.

let's say I have multiple pages with always a `title` and an `og:image` defined per page, like
`/pages/my-example-page.vue`
```
<script setup>
useSeoMeta({
  title: 'My page title'
  image: 'https://example.com/image.jpg'
})
</script>
```

So I have this result
```
<head
  <title>My page title</title>
  <meta property="og-image" content="https://example.com/image.jpg">
</head>
```

I would like to write a middleware so I can programmatically add other tags without having to add them page by page. So for example, I would like to end up with:

```
<head
  <title>My page title</title>
  <meta property="og:title" content="My page title">
  <meta property="og:image" content="https://example.com/image.jpg">
  <meta property="twitter:image" content="https://example.com/image.jpg">
</head>
```

For that I would need to check the metadata. What I'm trying to do is
`middleware/metaTags.global.ts`
```
export default defineNuxtRouteMiddleware (to => {
  if (to.meta?.title) {
    useSeoMeta({
      ogTitle: () => to.meta.title
      twitterImage: () => to.meta.ogImage
  }
 })
```

But this doesn't seem to be working as `to.meta` comes empty. How should I access the `useSeoMeta` data instead?

**Comment by danielroe** [comment]
to.meta (page meta data) and seo metadata are completely unrelated. You might be able to use an unhead hook to do what you want, however.

**Comment by Jesus82** [comment]
@danielroe @carlosvaldesweb I was able to achieve this with a composable (not with a middleware, as I just can access the head object before the page `useSeoMeta` updates them.

@harlan-zw do you think this is the best approach?

```
import { injectHead } from '@unhead/vue';

export const useMetaData = () => {
  const head = injectHead();

  function getMetaData() {
    console.log(head.headEntries().at(-1).input);
    return head.headEntries().at(-1).input;
  }

  function setMetaData() {
    useSeoMeta({
      ogTitle: getMetaData().title,
      twitterImage: getMetaData()._flatMeta.image,
    });
  }

  return { getMetaData, setMetaData };
};
```

Working example [here](https://stackblitz.com/edit/nuxt-starter-m4hv5q?file=app.vue,pages%2Findex.vue,middleware%2FmetaData.global.ts,composables%2FuseMetaData.ts)
